In [4]:
import pandas as pd
import matplotlib 
import plotly
import seaborn as sns
from pandas import Series,DataFrame
import csv

In [5]:
df=pd.read_csv('chengdupage_recommend.csv')

In [6]:
df

spot  recommend  rank                   satellite
0              都江堰景区      11462   4.7                  成都市都江堰市公园路
1               锦里古街      12771   4.4             成都市武侯区武侯祠大街231号
2               宽窄巷子      10069   4.4              成都市青羊区金河路口宽窄巷子
3              成都动物园       1849   4.6        成都市成华区昭觉寺南路234号(驷马桥)
4                青城山       5636   4.6            成都市都江堰市西南部青城山风景区
5        成都大熊猫繁育研究基地      12008   4.6           成都市成华区外北熊猫大道1375号
6                文殊院        837   4.5               成都市青羊区文殊院街66号
7               西岭雪山       5184   4.7            成都市大邑县西岭镇西岭雪山风景区
8           国色天乡水上乐园       1110   4.6         成都市温江区万春镇国色天乡二期水上世界
9                武侯祠       9108   4.5             成都市武侯区武侯祠大街231号
10            紫颐香薰山谷        520   4.1                   成都市双流县太平镇
11           金沙遗址博物馆       4454   4.7      成都市青羊区金沙遗址路2号成都金沙遗址博物馆
12              杜甫草堂       6238   4.5                成都市青羊区青华路37号
13               春熙路       1440   4.4                      成都市锦江区
14             成都欢乐谷       4043   4.6               成都市金牛区西华大道16号
15              人民公园        474   4.4                成都市青羊区少城路12号
16              青城后山        700   4.5               成都市都江堰市驿道街66号
17          国色天乡陆地乐园       1375   4.6           成都市温江区万春镇天乡路二段88号
18               黄龙溪        780   4.4                  成都市双流区黄龙溪镇
19               青羊宫        625   4.3              成都市青羊区一环路西二段九号
20           天堂岛海洋乐园       1514   4.6  成都市武侯区天府大道北段1700号环球中心东区负1层
21             四川博物院        408   4.5              成都市青羊区浣花南路251号
22              花舞人间       1838   4.5         成都市新津县永商镇梨花大道一段555号
23            童话森林乐园        303   4.5               成都市天府新区三星镇井石村
24        成都海昌极地海洋公园       3889   4.5        成都市双流区天府大道南沿线南端2039号
25              安仁古镇        293   4.4                   成都市大邑县迎宾路
26           刘氏庄园博物馆       1022   4.6             成都市大邑县安仁镇金桂街15号
27              洛带古镇        408   4.2                  成都市龙泉驿区洛带镇
28           建川博物馆聚落        782   4.7      成都市大邑县安仁镇迎宾路二段550建川博物馆
29          中国泰迪熊博物馆        771   4.3               成都市温江区金马国际体育城
..               ...        ...   ...                         ...
720            浣花夫人祠         27   4.3           成都市青羊区草堂路28号杜甫草堂内
721             诗书画院          8   4.9                     百花潭路10号
722               碑廊          8   4.4                        文殊院内
723              高堂寺          6   3.7                   成都市大邑县敦义乡
724            不屈战俘馆          1   4.0          成都市大邑县安仁古镇建川博物馆聚落内
725          飞派飞行俱乐部          2   5.0                成都市双流区黄龙溪大河村
726            老地方茶铺          4   4.5                   龙祥路2号附137
727            珍稀植物园          7   4.4                        植物园内
728             冯家大院          7   4.6            九思巷3号(骡马市地铁站A出口)
729   X记忆真人密室逃脱mini店          8   4.3     成都市成华区建设路第五大道星美沙河影城6楼6号
730          东较场特色街区          9   4.1                  东较场街38-12号
731  华侨城大剧院(纯水岸208区)          9   4.1                     西华大道16号
732             胡家大院          2   4.5                   成都市新津县金华镇
733          银爵舞蹈工作室         10   4.9        武侯区新会展中心南门成达佳园活动中心3楼
734            芒城古城址          3   4.0               成都市都江堰市青城乡芒城村
735             湖广会馆          8   4.5                    洛带镇下街迅峰旁
736           艺家国际画廊          7   4.7                   天和西三街330号
737              散花楼         21   4.6                  成都市青羊区锦里西路
738              千佛塔          5   3.6             成都市都江堰市幸福镇灵岩山南麓
739             润道茶社          5   4.2                   光华西三路156号
740             仙山花谷          3   3.7            成都市彭州市葛仙山万年场仙山花谷
741           艺林关美术馆          7   5.0                     琴台路135号
742             南站公园          9   4.1                        成汉北路
743            张采芹故居          9   4.3                      宽巷子35号
744       九又四分之三密室逃脱         11   4.8             春熙路王府井B座12楼12D号
745            川剧博物馆         10   4.5                      慈惠堂街8号
746             丽都公园         11   4.4                丽都路1号(丽都公园内)
747              飞仙阁          3   3.7                      成都市蒲江县
748          国学精品动物园          7   4.3     锦江大道大口潮火锅门口左边小

In [7]:
import plotly.plotly as py
import plotly.graph_objs as go
plotly.offline.init_notebook_mode(connected=True)
from plotly.graph_objs import Scatter, Layout
import numpy as np
from plotly.graph_objs import Bar
from plotly.offline import iplot

we extracted some kinds of attractions based on keywords provided by Chengdu local people from whose perspective these types can be representative of Chengdu's sceneries.

In [8]:
园林类=df['spot'].apply(lambda t:'园' in str(t)).aggregate('sum')
古镇=df['spot'].apply(lambda t:'古镇' in str(t)).aggregate('sum')
博物馆=df['spot'].apply(lambda t:'博物馆'in str(t)).aggregate('sum')
庭院=df['spot'].apply(lambda t:'院'in str(t)).aggregate('sum')
寺=df['spot'].apply(lambda t:'寺'in str(t)).aggregate('sum')
庙=df['spot'].apply(lambda t:'庙'in str(t)).aggregate('sum')
广场=df['spot'].apply(lambda t:'广场' in str(t)).aggregate('sum')

In [9]:
山=df['spot'].apply(lambda t:'山' in str(t)).aggregate('sum')
沟=df['spot'].apply(lambda t:'沟' in str(t)).aggregate('sum')
溪=df['spot'].apply(lambda t:'溪' in str(t)).aggregate('sum')
温泉=df['spot'].apply(lambda t:'温泉' in str(t)).aggregate('sum')
湖=df['spot'].apply(lambda t:'湖' in str(t)).aggregate('sum')

In [10]:
data = [go.Bar(
    x=['园林类', '古镇', '博物馆','庭院','寺','庙','广场'],
    y=[园林类,古镇,博物馆,庭院,寺,庙,广场],
    marker=dict(
        color=['rgba(222,45,38,0.8)', 'rgba(204,204,204,1)',
               'rgba(204,204,204,1)', 'rgba(204,204,204,1)',
               'rgba(204,204,204,1)','rgba(204,204,204,1)','rgba(204,204,204,1)']),
)]
iplot({
    "data": data,
    "layout": Layout(title="Human Landscape")
})

The most classic way to spend leisure time for Chengdu people is to sit in parks or gardens, chatting with family or friends, enjoying tea-tasting and playing Majiang. So parks and gardens as you can see make up the largest part of human landscape. 

In [11]:
data = [go.Bar(
    x=['山', '沟', '溪','温泉','湖'],
    y=[山,沟,溪,温泉,湖],
    marker=dict(
        color=['rgba(222,45,38,0.8)', 'rgba(204,204,204,1)',
               'rgba(204,204,204,1)', 'rgba(204,204,204,1)',
               'rgba(204,204,204,1)'])    
)]
iplot({
    "data": data,
    "layout": Layout(title="Natural Landscape"),
  
    
})

The landform of Chengdu is varied for its long and complicated tectonic activity，which also leads to diverse types of natural landscape.
Mountains contribute a large propotion to natural landscape.Being at the central of Sichuan Basin, Chengdu Plain is always a base camp for mountaineering enthusiasts,from where they can get to different kinds of mountains,including Mount Qingcheng,the birthplace of Taoism, the Xiling Snow Mountain that once described by Dufu,and also Minya Konka,which is the easternmost 7,000 metres (23,000 ft) peak in the world and be known as "the King of Sichuan Mountains".

In [12]:
df_top_100=df.sort_values(by='recommend',ascending=False)[:100].reset_index()

In [13]:
df_top_100

index             spot  recommend  rank  \
0       1             锦里古街      12771   4.4   
1       5      成都大熊猫繁育研究基地      12008   4.6   
2       0            都江堰景区      11462   4.7   
3       2             宽窄巷子      10069   4.4   
4       9              武侯祠       9108   4.5   
5      12             杜甫草堂       6238   4.5   
6       4              青城山       5636   4.6   
7       7             西岭雪山       5184   4.7   
8      11          金沙遗址博物馆       4454   4.7   
9      14            成都欢乐谷       4043   4.6   
10     24       成都海昌极地海洋公园       3889   4.5   
11    117      《芙蓉国粹》传奇变脸秀       2863   4.7   
12     37          蜀风雅韵川剧院       1935   4.8   
13      3            成都动物园       1849   4.6   
14     22             花舞人间       1838   4.5   
15     36              石象湖       1693   4.6   
16     20          天堂岛海洋乐园       1514   4.6   
17     13              春熙路       1440   4.4   
18     17         国色天乡陆地乐园       1375   4.6   
19     52            南湖梦幻岛       1342   4.6   
20     39         国色天乡童话世界       1157   4.4   
21      8         国色天乡水上乐园       1110   4.6   
22    135             秀丽东方       1069   4.6   
23     26          刘氏庄园博物馆       1022   4.6   
24      6              文殊院        837   4.5   
25     28          建川博物馆聚落        782   4.7   
26     18              黄龙溪        780   4.4   
27     29         中国泰迪熊博物馆        771   4.3   
28     16             青城后山        700   4.5   
29     19              青羊宫        625   4.3   
..    ...              ...        ...   ...   
70    144            成都植物园        199   4.3   
71     51          三圣花乡旅游区        198   4.3   
72    217      大梁酒庄田园乐翻天乐园        197   4.5   
73    177             蔚然花海        196   4.1   
74    253             虹口漂流        190   4.5   
75     67            塔子山公园        189   4.3   
76    155      新都尖锋旱雪四季滑雪场        188   4.9   
77     88            郫县望丛祠        186   4.3   
78    237  川西竹海峡谷自然景区（金鸡谷）        176   4.2   
79    254          生命奥秘博物馆        176   4.6   
80    128             成都映象        174   4.6   
81    192          成都映象看川剧        165   4.3   
82    133             琴台故径        163   4.2   
83    140              文殊坊        159   4.4   
84     49              上清宫        158   4.4   
85    262            星期8小镇        155   4.6   
86    233         花水湾四季温泉谷        153   4.4   
87    137           西南民族大学        153   4.6   
88    190         汤乐源温泉体验馆        150   4.5   
89     59              九眼桥        150   4.7   
90    228           狐巴巴冒险园        148   4.9   
91     55              天师洞        142   4.4   
92    191              白鹿镇        139   4.1   
93     66              井巷子        139   4.4   
94     56            百花潭公园        138   4.6   
95    292        3D魔幻/魔画世界        138   4.0   
96    169              丹景山        128   4.3   
97    315             棠湖公园        126   4.6   
98    164             新华公园        125   4.3   
99    157             温江公园        125   4.5   

                                 satellite  
0                          成都市武侯区武侯祠大街231号  
1                        成都市成华区外北熊猫大道1375号  
2                               成都市都江堰市公园路  
3                           成都市青羊区金河路口宽窄巷子  
4                          成都市武侯区武侯祠大街231号  
5                             成都市青羊区青华路37号  
6                         成都市都江堰市西南部青城山风景区  
7                         成都市大邑县西岭镇西岭雪山风景区  
8                   成都市青羊区金沙遗址路2号成都金沙遗址博物馆  
9                            成都市金牛区西华大道16号  
10                    成都市双流区天府大道南沿线南端2039号  
11                 成都市锦江区华兴正街54号（近王府井）锦江剧场  
12            成都市青羊区琴台路132号市文化公园门口蜀风雅韵民俗茶馆  
13                    成都市成华区昭觉寺南路234号(驷马桥)  
14                     成都市新津县永商镇梨花大道一段555号  
15                     成都市蒲江县成雅高速公路86公里出口处  
16              成都市武侯区天府大道北段1700号环球中心东区负1层  
17                                  成都市锦江区  
18                       成都市温江区万春镇天乡路二段88号  
19                        成都市双流县南延线南湖度假风景区  
20                      成都市温江区万春镇88号国色天乡乐园  
21                     成都市温江区万春镇国色天乡二期水上世界  
22                     成都市锦江区锦江大道湿地路秀丽东方园区  
23       

In [14]:
data = [go.Scatter(
    x = df_top_100.index,
    y = df_top_100['rank'].values,
    mode = 'lines+markers',
    name = 'recommend'   
)]
iplot({"data":data, 
       "layout": Layout(title="scores of comments_top_100 spot")
      })

we sorted all scenic spots from high to low according to the number of comments,finding out the top 100 most popular spots in Chengdu, within which we continue to analysis the quality of scenic spots by checking scores.  
We find out that not all spots with a large amount of comments get high scores at the same time. 

In [15]:
top_100_recommend=df_top_100[df_top_100['rank']>=4.7]

In [16]:
top_100_recommend.sort_values(by='rank',ascending=False)

index         spot  recommend  rank                     satellite
76    155  新都尖锋旱雪四季滑雪场        188   4.9       成都市新都区北星干线成德大道马家镇北星小区对面
90    228       狐巴巴冒险园        148   4.9            成都市都江堰市莲花湖旅游公路188号
12     37      蜀风雅韵川剧院       1935   4.8  成都市青羊区琴台路132号市文化公园门口蜀风雅韵民俗茶馆
65    183    国色天乡尖叫大马戏        211   4.8               成都市温江区万春镇国色天乡乐园
2       0        都江堰景区      11462   4.7                    成都市都江堰市公园路
7       7         西岭雪山       5184   4.7              成都市大邑县西岭镇西岭雪山风景区
8      11      金沙遗址博物馆       4454   4.7        成都市青羊区金沙遗址路2号成都金沙遗址博物馆
11    117  《芙蓉国粹》传奇变脸秀       2863   4.7       成都市锦江区华兴正街54号（近王府井）锦江剧场
25     28      建川博物馆聚落        782   4.7        成都市大邑县安仁镇迎宾路二段550建川博物馆
41    149         通济花海        355   4.7              彭州市通济镇乔楼村湔江路通济花海
44     48        浣花溪公园        315   4.7            成都市青羊区青华路9号(近杜甫草堂)
56    145      国色天乡摩天轮        250   4.7             成都市温江区万春镇天乡路二段88号
89     59          九眼桥        150   4.7                    成都市武侯区望江路上

Apart from those well-known spots like Dujiangyan and Xiling Snow Mountain,we introduce some sleepers of Chengdu attractions which may not be very famous but you still can have a goodtime there（scores higher than 4.7), such as the ski resort located in Xindu(4.9) and the Shufenyayun Theatre where you can enjoy Sichuan Opera(4.8 and 1935comments).